app/code/Magento/Sitemap/Model/Observer.php

```diff
@@ -67,24 +69,33 @@ class Observer
     protected $inlineTranslation;
 
     /**
+     * @var \Magento\Store\Model\App\Emulation $appEmulation
+     */
+    private $appEmulation;
+
+    /**
+     * Observer constructor.
      * @param \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig
-     * @param \Magento\Sitemap\Model\ResourceModel\Sitemap\CollectionFactory $collectionFactory
+     * @param ResourceModel\Sitemap\CollectionFactory $collectionFactory
      * @param \Magento\Store\Model\StoreManagerInterface $storeManager
      * @param \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder
      * @param \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation
+     * @param Emulation|null $appEmulation
      */
     public function __construct(
         \Magento\Framework\App\Config\ScopeConfigInterface $scopeConfig,
-        \Magento\Sitemap\Model\ResourceModel\Sitemap\CollectionFactory $collectionFactory,
+        ResourceModel\Sitemap\CollectionFactory $collectionFactory,
         \Magento\Store\Model\StoreManagerInterface $storeManager,
         \Magento\Framework\Mail\Template\TransportBuilder $transportBuilder,
-        \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation
+        \Magento\Framework\Translate\Inline\StateInterface $inlineTranslation,
+        Emulation $appEmulation
```

Comment:
This new parameter should be optional

Reply:
Same here if i add optional, as this have been in older commit the codeMess test  will fail.
`Backward Compatibility Policy is not applied to Plugins, Observers and Setup Scripts.`

Reply:
`The class Observer has a coupling between objects value of 13.`